### ISSUE TITLE
Investigate use of Vélin for autoreformatting docstrings to be numpydoc style

### ISSUE THREAD

**Issue by matthewfeickert** [scikit-hep/pyhf#942]
# Description

@Carreau's at it again making wonderful tools with [Vélin](https://github.com/Carreau/minirst) which can autoreformat docstrings to be numpydoc style! This could be potentially _very_ useful for application in CI and in the docs creation process.

@bdice has [already used `velin` to lint and reformat](https://twitter.com/signacdata/status/1282136263626240000) the [`signac`](https://github.com/glotzerlab/signac) docs so can offer some views as well.

**Comment by Carreau** [comment]
Note that this is not using numpydoc format, so it will likely have little if no effects.

**Comment by matthewfeickert** [comment]
> Note that this is not using numpydoc format, so it will likely have little if no effects.

Yup, this is true as I copied PyTorch back when we started our docs. Though having a tool like yours might make switching far easier.

**Comment by Carreau** [comment]
I'm happy to get a PR that allow to parse the pytorch doc format into a numpydoc.docscrape.NumpyDocstring object, if I have that then generating a new docstring and replacing it is relatively straitforward.

**Comment by matthewfeickert** [comment]
> I'm happy to get a PR that allow to parse the pytorch doc format into a numpydoc.docscrape.NumpyDocstring object, if I have that then generating a new docstring and replacing it is relatively straitforward.

Great idea! That won't happen from me this next week given that I'm co-running a conference, but I like this idea very much. Forking now so that I have a reminder.
